### PR TITLE
Branch off of v2.5 - using self:log instead of NSLog

### PR DESCRIPTION
**Pull request by wmathurin**

### CODE REVIEW

File: src/fmdb/FMDatabase.m [hunk 1]
```diff
@@ -136,7 +136,7 @@ - (BOOL)open {
     
     int err = sqlite3_open([self sqlitePath], &_db );
     if(err != SQLITE_OK) {
-        NSLog(@"error opening!: %d", err);
+        [self log:SFLogLevelDebug format:@"error opening!: %d", err];
```
Comment (khawkins) [inline src/fmdb/FMDatabase.m]: Should errors be logged at `SFLogLevelError`?

Comment (wmathurin) [inline src/fmdb/FMDatabase.m]: My quick replace was too quick.
I'll use error/warning/debug wherever appropriate

**Comment by wmathurin** [comment]
Adjusted log levels with https://github.com/wmathurin/fmdb/commit/99a5c6e73098afb72d301bfe709fa0217f43659a

File: src/fmdb/FMDatabase.m [hunk 2]
```diff
@@ -234,7 +234,7 @@ static int FMDBDatabaseBusyHandler(void *f, int count) {
         int requestedSleepInMillseconds = arc4random_uniform(50) + 50;
         int actualSleepInMilliseconds = sqlite3_sleep(requestedSleepInMillseconds);
         if (actualSleepInMilliseconds != requestedSleepInMillseconds) {
-            NSLog(@"WARNING: Requested sleep of %i milliseconds, but SQLite returned %i. Maybe SQLite wasn't built with HAVE_USLEEP=1?", requestedSleepInMillseconds, actualSleepInMilliseconds);
+            [self log:SFLogLevelwarning format:@"WARNING: Requested sleep of %i milliseconds, but SQLite returned %i. Maybe SQLite wasn't built with HAVE_USLEEP=1?", requestedSleepInMillseconds, actualSleepInMilliseconds];
```
Comment (khawkins) [inline src/fmdb/FMDatabase.m]: This should be `SFLogLevelWarning`, to match the enum.

File: src/fmdb/FMDatabase.m [hunk 3]
```diff
@@ -454,7 +454,7 @@ - (BOOL)databaseExists {
     
     if (!_db) {
             
-        NSLog(@"The FMDatabase %@ is not open.", self);
+        [self log:SFLogLevelDebug format:@"The FMDatabase %@ is not open.", self];
```
Comment (khawkins) [inline src/fmdb/FMDatabase.m]: This looks like an error case: the app is going to crash if configured to do so.

File: src/fmdb/FMDatabase.m [hunk 4]
```diff
@@ -1089,8 +1089,8 @@ - (BOOL)executeUpdate:(NSString*)sql error:(NSError**)outErr withArgumentsInArra
     
     if (closeErrorCode != SQLITE_OK) {
         if (_logsErrors) {
-            NSLog(@"Unknown error finalizing or resetting statement (%d: %s)", closeErrorCode, sqlite3_errmsg(_db));
-            NSLog(@"DB Query: %@", sql);
+            [self log:SFLogLevelDebug format:@"Unknown error finalizing or resetting statement (%d: %s)", closeErrorCode, sqlite3_errmsg(_db)];
```
Comment (khawkins) [inline src/fmdb/FMDatabase.m]: This also looks like an error (failed to close the DB).

File: src/fmdb/FMResultSet.m [hunk 5]
```diff
@@ -188,7 +188,7 @@ - (BOOL)nextWithError:(NSError **)outErr {
     }
     else {
         // wtf?
-        NSLog(@"Unknown error calling sqlite3_step (%d: %s) rs", rc, sqlite3_errmsg([_parentDB sqliteHandle]));
+        [self log:SFLogLevelDebug format:@"Unknown error calling sqlite3_step (%d: %s) rs", rc, sqlite3_errmsg([_parentDB sqliteHandle])];
```
Comment (khawkins) [inline src/fmdb/FMResultSet.m]: This also looks like an error case.

**Comment by khawkins** [comment]
Otherwise, looks fine to me.

**Comment by bhariharan** [comment]
:+1:

**Pull request merged**